### PR TITLE
Add change log for 0.4.0

### PR DESCRIPTION
**Pull request by JDGrimes**

### CODE REVIEW

**Comment by JDGrimes** [comment]
Build failures are probably Travis's fault: https://twitter.com/traviscistatus/status/592902357144498176

They haven't rolled back the changes on the container-based infrastructure yet: https://www.traviscistatus.com/incidents/h91j8ft0qs2l

After they do, we can re-run the failed builds and I believe they will succeed.

**Comment by GaryJones** [comment]
The failure on HHVM may well be Travis's fault, but [5.3](https://travis-ci.org/WordPress-Coding-Standards/WordPress-Coding-Standards/jobs/60376472) tests and 5.4 look to be down to unit test failures.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,52 @@ This projects adheres to [Semantic Versioning](http://semver.org/) and [Keep a C
...
+- `Squiz.ControlStructures.ControlSignature` sniff to the rulesets.
+
+### Changed
+- `WordPress.Arrays.ArrayKeySpacingRestrictions` sniff to give errors for `NoSpacesAroundArrayKeys` and `SpacesAroundArrayKeys` instead of just warnings.
+- `WordPress.NamingConventions.ValidFunctionName` sniff to allow for camel caps method names in child classes.
+- `WordPress.XSS.EscapeOutput` sniff to allow for integers (e.g. `echo 5` and `print( -1 )`).
+
+### Removed
+- Errors for mixed key/keyless array elements in the `WordPress.Arrays.ArrayDeclaration` sniff.
+- BOM from `WordPress.WhiteSpace/OperatorSpacing` sniff file.
```
Comment (GaryJones) [inline CHANGELOG.md]: Is this `/` supposed to be a `.`, or is the `.` supposed to be a `/`, or neither?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,7 +6,52 @@ This projects adheres to [Semantic Versioning](http://semver.org/) and [Keep a C
...
+### Fixed
+- Incorrect errors for `else` statements using alternative syntax.
+- `WordPress.VIP.ValidatedSanitizedInput` sniff not always treating casting as sanitization.
+- `WordPress.XSS.EscapeOutput` sniff flagging comments as needing to be escaped.
+- `WordPress.XSS.EscapeOutput` sniff not sniffing comma-delimited `echo` arguments after encountering the first escaping function in the statement.
+- `WordPress.PHP.YodaConditions` sniff not flagging comparisons to constants or function calls.
+- `WordPress.Arrays.ArrayDeclaration` sniff not ignoring doc comments.
+- Link to phpStorm instructions in `README.md`.
+- Poor performance of the `WordPress.Arrays.ArrayAssignmentRestrictions` sniff.
+- Poor performance of teh `WordPress.Files.FileName` sniff.
```
Comment (GaryJones) [inline CHANGELOG.md]: Typo "teh".

**Comment by GaryJones** [comment]
Couple of comments, but fine to merged when addressed.

File: CHANGELOG.md (hunk 1 above)
Comment (JDGrimes) [inline CHANGELOG.md]: It's supposed to be a `.`.

**Comment by GaryJones** [comment]
+1 one to merge.

**Comment by JDGrimes** [comment]
OK. I'm going to wait to merge until Travis has finished rolling out the revert so we can see if these errors are genuine or not.

**Comment by JDGrimes** [comment]
This is now passing.

**Pull request merged**